crates/turborepo-scm/src/git.rs

```diff
@@ -81,6 +85,45 @@ impl SCM {
     }
 }
 
+const UNKNOWN_SHA: &str = "0000000000000000000000000000000000000000";
+
+#[derive(Debug, Deserialize, Clone)]
+struct GitHubCommit {
+    id: String,
+}
+
+#[derive(Debug, Deserialize, Default)]
+struct GitHubEvent {
+    #[serde(default)]
+    before: String,
+
+    #[serde(default)]
+    commits: Vec<GitHubCommit>,
+}
+
+impl GitHubEvent {
+    fn get_first_commit_id(&self) -> Option<String> {
+        if self.commits.is_empty() {
+            // commits can be empty when you push a branch with no commits
+            return None;
+        }
+
+        if self.commits.len() > 2048 {
+            // GitHub API limit for number of commits shown in this field
+            return None;
+        }
+
+        // Extract the base ref from the push event
+        if let Some(first_commit) = self.commits.first() {
+            let id = &first_commit.id;
+            return Some(format!("{id}^"));
+        }
+
+        // we don't have enough information to determine the base ref
+        None
```

Comment:
Just a small change, `?` works on `Option` too
```suggestion
        let first_commit = self.commits.first()?; 
        let id = &first_commit.id;
        Some(format!("{id}^"))
```

Reply:
oh sweet, great find! yeah, I was using `?` a lot on the base function where this code was extracted from (`get_github_base_ref`).